spot_ros2_control/test/pytests/test_spot_ros2_control.py

```diff
@@ -0,0 +1,73 @@
+import itertools
+
+import pytest
+from bosdyn.api.robot_command_pb2 import RobotCommandResponse
+from bosdyn.api.robot_state_pb2 import RobotStateStreamResponse
+from sensor_msgs.msg import JointState
+from synchros2.futures import wait_for_future
+from synchros2.scope import ROSAwareScope
+from synchros2.subscription import Subscription
+
+from spot_wrapper.testing.fixtures import SpotFixture
+
+LEG_JOINTS = [
+    "front_left_hip_x",
+    "front_left_hip_y",
+    "front_left_knee",
+    "front_right_hip_x",
+    "front_right_hip_y",
+    "front_right_knee",
+    "rear_left_hip_x",
+    "rear_left_hip_y",
+    "rear_left_knee",
+    "rear_right_hip_x",
+    "rear_right_hip_y",
+    "rear_right_knee",
+]
+ARM_JOINTS = ["arm_sh0", "arm_sh1", "arm_el0", "arm_el1", "arm_wr0", "arm_wr1", "arm_f1x"]
+
+NUM_JOINTS_ARM = 19
+
+
+def state_stream_response(position: list[float], velocity: list[float], load: list[float]) -> RobotStateStreamResponse:
+    response = RobotStateStreamResponse()
+    response.joint_states.position.extend(position)
+    response.joint_states.velocity.extend(velocity)
+    response.joint_states.load.extend(load)
+    response.inertial_state.packets.add()  # hardware interface expects at least a packet
+    response.contact_states.extend([0] * 4)  # hardware interface expects all 4 feet to have a contact state
+    return response
+
+
+@pytest.mark.launch(fixture=pytest.robot_spot_ros2_control)
+def test_joint_states(simple_spot: SpotFixture, ros: ROSAwareScope) -> None:
+    # Mock a joint state stream response
+    position = list(range(0, NUM_JOINTS_ARM))
+    velocity = list(range(NUM_JOINTS_ARM, 2 * NUM_JOINTS_ARM))
+    load = list(range(2 * NUM_JOINTS_ARM, 3 * NUM_JOINTS_ARM))
+    stream_response = state_stream_response(position=position, velocity=velocity, load=load)
+    simple_spot.api.GetRobotStateStream.future.returns(itertools.repeat(stream_response))
+    response = RobotCommandResponse()
+    response.status = RobotCommandResponse.Status.STATUS_OK
+    simple_spot.api.RobotCommand.future.returns(response)
+    call = simple_spot.api.JointControlStream.serve(timeout=2.0)
+    assert call is not None
+    spot_name = simple_spot.api.name
+    # Grab the latest joint states topic
+    joint_states = Subscription(JointState, f"/{spot_name}/low_level/joint_states", node=ros.node)
+    assert wait_for_future(joint_states.update, timeout_sec=10.0)
+    joint_state_message = joint_states.latest
+    assert joint_state_message is not None
```

Comment:
@khughes-bdai nit: you can collapse these three into: 
```suggestion
    joint_state_message = unwrap_future(joint_states.update, timeout_sec=10.0)
```

Reply:
Added! Thanks